ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/builders.kt

```diff
@@ -30,26 +31,28 @@ public fun HttpClientConfig<*>.SSE(config: SSEConfig.() -> Unit) {
 /**
  * Opens a [ClientSSESession].
  */
-public suspend fun HttpClient.serverSentEventsSession(
+public suspend fun <T : Any> HttpClient.serverSentEventsSession(
+    deserialize: ((String) -> T)? = null,
     reconnectionTime: Duration? = null,
     showCommentEvents: Boolean? = null,
     showRetryEvents: Boolean? = null,
     block: HttpRequestBuilder.() -> Unit
-): ClientSSESession {
+): ClientSSESession<T> {
```

Comment:
Did you consider using overloads instead of the default deserialize argument here to allow the original API sans type argument when using strings?

By that, I mean something like...

```
public suspend fun HttpClient.sseSession(
    reconnectionTime: Duration? = null,
    showCommentEvents: Boolean? = null,
    showRetryEvents: Boolean? = null,
    block: HttpRequestBuilder.() -> Unit
) = sseSession({ it }, reconnectionTime, showCommentEvents, showRetryEvents, block)

public suspend fun <T : Any> HttpClient.sseSession(
    deserialize: (String) -> T,
    reconnectionTime: Duration? = null,
    showCommentEvents: Boolean? = null,
    showRetryEvents: Boolean? = null,
    block: HttpRequestBuilder.() -> Unit
)
```

This would make it so the default deserialization can't be used, however.

Reply:
I think maybe ignore this suggestion and just include `typeInfo` in the default lambda signature.  We'll probably want some logic to ensure that when `String` is supplied as the type argument that we avoid any serialization, otherwise you'd end up with some weirdness with it expecting strings to be wrapped in quotes.